master/internal/usergroup/authz_iface.go

```diff
@@ -0,0 +1,28 @@
+package usergroup
+
+import (
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/authz"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// UserGroupAuthZ describes authz methods for `user` package.
+type UserGroupAuthZ interface {
+	// CanGetGroup checks whether a user can get a group.
+	// GET /api/v1/groups/{group_id}
+	CanGetGroup(curUser model.User, gid int) (bool, error)
+
+	// FilterGroupsList checks what groups a user can get.
+	// POST /api/v1/groups/search
+	FilterGroupsList(curUser model.User, query *bun.SelectQuery) (*bun.SelectQuery, error)
+
+	// CanUpdateGroups checks if a user can create, delete, or update a group.
+	// POST /api/v1/groups
+	// PUT /api/v1/groups/{group_id}
+	// DELETE /api/v1/groups/{group_id}
+	CanUpdateGroups(curUser model.User) (bool, error)
```

Comment:
I think we should add `ctx context.Context` as the first parameter of these functions. The basic version will ignore it, but the RBAC version (and others that need to e.g. make database calls) will use it instead of having to do `context.TODO`.